src/main/java/uk/gov/hmcts/reform/sscs/util/SscsUtil.java

```diff
@@ -43,24 +45,32 @@ public static boolean isValidCaseState(State state, List<State> allowedStates) {
         return allowedStates.contains(state);
     }
 
+    public static void clearPostHearingFields(SscsCaseData caseData) {
+        caseData.setPostHearing(new PostHearing());
+        clearDocumentTransientFields(caseData);
+    }
+
     public static void clearDocumentTransientFields(SscsCaseData caseData) {
         caseData.setDocumentGeneration(DocumentGeneration.builder().build());
         caseData.setDocumentStaging(DocumentStaging.builder().build());
     }
 
     public static void addDocumentToDocumentTab(FooterService footerService, SscsCaseData caseData, DocumentType documentType) {
         DocumentLink url = caseData.getDocumentStaging().getPreviewDocument();
-        String now = LocalDate.now().format(DateTimeFormatter.ofPattern("dd-MM-yyyy"));
-        SscsDocumentTranslationStatus documentTranslationStatus = null;
-        if (caseData.isLanguagePreferenceWelsh()) {
-            documentTranslationStatus = SscsDocumentTranslationStatus.TRANSLATION_REQUIRED;
-        }
-        footerService.createFooterAndAddDocToCase(url, caseData, documentType, now,
-            null, null, documentTranslationStatus);
-        if (documentTranslationStatus != null) {
-            caseData.setInterlocReviewState(InterlocReviewState.WELSH_TRANSLATION);
-            log.info("Set the InterlocReviewState to {},  for case id : {}", caseData.getInterlocReviewState(), caseData.getCcdCaseId());
-            caseData.setTranslationWorkOutstanding(YesNo.YES.getValue());
+
+        if (nonNull(url)) {
+            String now = LocalDate.now().format(DateTimeFormatter.ofPattern("dd-MM-yyyy"));
+            SscsDocumentTranslationStatus documentTranslationStatus = null;
+            if (caseData.isLanguagePreferenceWelsh()) {
+                documentTranslationStatus = SscsDocumentTranslationStatus.TRANSLATION_REQUIRED;
+            }
+            footerService.createFooterAndAddDocToCase(url, caseData, documentType, now,
```

Comment:
Do we ever need to pass in a null url here downstream? I'm all for null checking but if it's a valid case to send null then we won't need the if.

Reply:
So not all the post hearings journeys will get the preview document screen, as of this it can be null sometimes so thats why ive added in the check

Reply:
Cool, I mean if for whatever reason we do want to pass null here we can always change it in a later ph journey.